modules/Microsoft.App/containerApps/deploy.bicep

```diff
@@ -0,0 +1,241 @@
+    }
+  }
+}
+
+@description('Optional. Custom domain bindings for Container App hostnames.')
+param customDomains array = []
+
+@description('Optional. Exposed Port in containers for TCP traffic from ingress.')
+param exposedPort int = 0
+
+@description('Optional. Rules to restrict incoming IP address.')
+param ipSecurityRestrictions array = []
+
+@description('Optional. Associates a traffic label with a revision. Label name should be consist of lower case alphanumeric characters or dashes.')
+param trafficLabel string = 'label-1'
+
+@description('Optional. Indicates that the traffic weight belongs to a latest stable revision.')
+param trafficLatestRevision bool = true
+
+@description('Optional. Name of a revision.')
+param trafficRevisionName string = ''
+
+@description('Optional. Traffic weight assigned to a revision.')
+param trafficWeight int = 100
+
+@description('Optional. Dapr configuration for the Container App.')
+param dapr object = {}
+
+@description('Optional. Max inactive revisions a Container App can have.')
+param maxInactiveRevisions int = 0
+
+@description('Optional. Container start command arguments.')
+param containerArgs array = []
+
+@description('Optional. Container start command.')
+param containerStartCommand array = []
+
+@description('Optional. List of probes for the container.')
+param containerAppProbe array = []
+
+@description('Optional. Container volume mounts.')
+param containerVolumeMounts array = []
+
+@description('Optional. List of specialized containers that run before app containers.')
+param initContainersTemplate array = []
+
+@description('Optional. User friendly suffix that is appended to the revision name.')
+param revisionSuffix string = ''
+
+@description('Optional. List of volume definitions for the Container App.')
+param volumes array = []
+
+@description('Optional. Workload profile type to pin for container app execution.')
+param workloadProfileType string = ''
+
+resource containerApps 'Microsoft.App/containerApps@2022-06-01-preview' = {
+  name: name
+  tags: tags
+  location: location
+  identity: identity
+  properties: {
+    environmentId: environmentId
+    configuration: {
+      activeRevisionsMode: activeRevisionsMode
+      dapr: !empty(dapr) ? dapr : null
+      ingress: {
+        allowInsecure: ingressAllowInsecure
+        customDomains: !empty(customDomains) ? customDomains : null
+        exposedPort: exposedPort
+        external: ingressExternal
+        ipSecurityRestrictions: !empty(ipSecurityRestrictions) ? ipSecurityRestrictions : null
+        targetPort: ingressTargetPort
+        traffic: [
+          {
+            label: trafficLabel
+            latestRevision: trafficLatestRevision
+            revisionName: trafficRevisionName
+            weight: trafficWeight
+          }
+        ]
+        transport: ingressTransport
+      }
+      maxInactiveRevisions: maxInactiveRevisions
+      registries: !empty(registries) ? registries : null
+      secrets: []
```

Comment:
Are we purposely avoiding this because it can be passed as plain text?

```bicep
      secrets: [
        {
          name: 'string'
          value: 'string'
        }
      ]
```

Reply:
God knows. I'll generate a value - I guess